### ISSUE TITLE
Table component could be more flexible with the columns

### ISSUE THREAD

**Issue by Dagrol** [pivotal-cf/pivotal-ui#414]
This is more of a feature request than an issue. Currently there is no way to enhance column headers with nested objects. For example, there is no way to add a Tooltip component within the column header. Is there is a neat way to achieve this or is it just not possible?

Thanks

**Comment by charleshansen** [comment]
Looking at the code, the `displayName` property in the `columns` property will accept a React component itself (in the documentation, we say it has to be a string). I think if you give the `Table` columns that look like 

```js
columns = [
{attribute: 'title', displayName: <OverlayTrigger ....(your tooltip code here)/>}
]
```
it should render a tooltip. This should work for any custom components as well. The one gotcha is the Table component won't inject any props into the header for you. Let me know if it works for you, no one has really wanted to customize the header yet.

**Comment by Dagrol** [comment]
Tried setting the displayName property to the OverlayTrigger component. It doesn't seem to like it. It either defaults to the attribute name or throw an exception that a string is expected for the displayName.

**Comment by charleshansen** [comment]
I just tried it myself and it seems to work. The column I created looked was
```js
{
        attribute: 'instances',
        sortable: true,
        displayName: (<OverlayTrigger placement="right" overlay={<Tooltip>I should be on the right</Tooltip>}>
          <span className="overlay-trigger" tabIndex="0"> tooltip on the right.</span>
        </OverlayTrigger>)
}
```

What version of pui-react-table are you using? If it is less than 5.0, table is not as usable. If you are using version 5+, I'm curious who is actually throwing that string error, since I can't find it in the pui table code.

**Comment by Dagrol** [comment]
Just tried that again and am getting the same error as before:
<img width="836" alt="warning" src="https://cloud.githubusercontent.com/assets/15840450/17848704/cf12d920-684c-11e6-8527-9067889ce2a8.png">

Here is how I created the column:

```
{
        attribute: 'crossCheck',
        sortable: true,
        displayName: (<OverlayTrigger placement="right" overlay={<Tooltip>I should be on the right</Tooltip>}>
            <span className="overlay-trigger" tabIndex="0"> tooltip on the right.</span>
        </OverlayTrigger>),
        headerProps: {
            className: ourHeaderClassName,
        },
        CustomCell: MediumCell
};
```

I am using version 5.3.1.

**Comment by charleshansen** [comment]
Ah, that error you are getting is because one of your React Components isn't correctly defined at rendering time. Based on the code I am looking at, it is `OverlayTrigger`, `Tooltip`, or `MediumCell`.

One of the more common ways to get in this place (assuming you are using ES6) is to use
`const OverlayTrigger = require('pui-react-overlay-trigger').OverlayTrigger;  //invalid`
instead of 
`const {OverlayTrigger} = require('pui-react-overlay-trigger').OverlayTrigger; //valid`

The invalid case will assign `OverlayTrigger` variable to an object that contains a react component instead of a react component itself. You won't get any errors about this except for the "type should not be null,..." error.

**Comment by Dagrol** [comment]
It is definitely the OverlayTrigger component. I've tried declaring it like you mentioned and removed the Tooltip entirely to have something like:
```
displayName: (<OverlayTrigger placement="right"> </OverlayTrigger>)
```
That still throws the same error and fails to render the page. I've also tried a global import:
```
import { OverlayTrigger } from 'pui-react-overlay-trigger';
```
When I set the displayName to a string, it seems to work fine. Pretty sure I am missing something very subtle.
